src/modal/modal-window.ts

```diff
@@ -76,4 +82,15 @@ export class NgbModalWindow implements OnInit,
     this._elWithFocus = null;
     this._renderer.setElementClass(document.body, 'modal-open', false);
   }
+
+  private _setAriaLabelledByToModal(): void {
+    const modalHeader: HTMLHeadingElement = this._elRef.nativeElement.getElementsByClassName('modal-title')[0];
+    if (modalHeader) {
+      const modalHeaderId: string = modalHeader.getAttribute('id') === null || modalHeader.getAttribute('id') === '' ?
+          `modal-header-${nextId++}` :
+          modalHeader.getAttribute('id');
+      this._renderer.setElementAttribute(modalHeader, 'id', modalHeaderId);
+      this._renderer.setElementAttribute(this._elRef.nativeElement, 'aria-labelledby', modalHeaderId);
+    }
+  }
```

Comment:
Personally I don't like this kind of DOM manipulations but, since the modal content can be everything I had no better ideas.
Please let me know if there will be any update / ideas on this cause I am really interested in this. 

Thanks.